lib/hardware/modules/div/iob_div_subshift_frac/iob_div_subshift_frac.py

```diff
@@ -13,22 +22,196 @@ def setup(py_params_dict):
+                ],
             },
             {
-                "core_name": "iob_reg_e",
-                "instance_name": "iob_reg_e_inst",
+                "name": "res_acc",
+                "descr": "res_acc wire",
+                "signals": [
+                    {"name": "res_acc", "width": "DATA_W+1"},
+                ],
+            },
+            {
+                "name": "pc",
+                "descr": "pc wire",
+                "signals": [
+                    {"name": "pc", "width": 2},
+                ],
+            },
+            {
+                "name": "div_frac",
+                "descr": "Division interface",
+                "signals": [
+                    {
+                        "name": "dividend",
+                    },
+                    {
+                        "name": "divisor",
+                    },
+                    {
+                        "name": "quotient_int",
+                    },
+                    {
+                        "name": "remainder",
+                    },
+                ],
+            },
+        ],
+        "blocks": [
+            {
+                "core_name": "iob_reg",
+                "instance_name": "divisor_reg0",
+                "parameters": {
+                    "DATA_W": "DATA_W",
+                    "RST_VAL": "1'b0",
+                },
+                "connect": {
+                    "clk_en_rst": "clk_en_rst",
+                    "data_i": "divisor_i",
+                    "data_o": "divisor_reg",
+                },
             },
             {
                 "core_name": "iob_div_subshift",
-                "instance_name": "iob_div_subshift_inst",
+                "instance_name": "div_subshift0",
+                "parameters": {
+                    "DATA_W": "DATA_W",
+                },
+                "connect": {
+                    "clk_en_rst": "clk_en_rst",
+                    "div": "div_frac",
+                },
             },
         ],
+        "combs": [
+            {
+                "verilog_code": """
+            incr        = 1'b0;
+            res_acc_nxt = res_acc + remainder_o;
+            res_acc_en  = 1'b0;
+            pc_nxt      = pc + 1'b1;
+                """,
+            }
+        ],
+        "fsms": [
+            {
+                "verilog_code": """
+        idle:
+
+        #case 0
+            if (!start_i) begin
+                pc_nxt = pc; 
+            end 
+
+        #case 1
+            if (!done_o) begin
+                pc_nxt = pc;
+            end
+        
+        #default
+            begin 
+            res_acc_en = 1'b1;
+            if (res_acc_nxt >= divisor_i) begin
+            incr        = 1'b1;
+            res_acc_nxt = res_acc + remainder_o - divisor_i;
+            end
+            if (!start_i) pc_nxt = pc;
+            else pc_nxt = 1'b1;
+            end
+
+        #case pc   
+            begin   
+            pc_nxt = idle; end
+            """,
```

Comment:
What is this for?
From the contents of the other states, I don't think `pc` can reach this sate.